### PR TITLE
GH-3463 upgrade junit to 5.8.1

### PR DESCRIPTION
**Pull request by hmottestad**
GitHub issue resolved: #3463  

Briefly describe the changes proposed in this PR:

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md) for more details):

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [ ] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Comment by hmottestad** [comment]
A few tests are using random data, which means that if they fail then there is no way to reproduce that exact run locally. Hopefully this will be better with fixed seeds.

File: testsuites/repository/src/main/java/org/eclipse/rdf4j/repository/TupleQueryResultTest.java [hunk 1]
```diff
@@ -232,8 +233,8 @@ public void testNotClosingResult() {
 				TupleQuery tupleQuery = repCon.prepareTupleQuery(QueryLanguage.SPARQL, queryString);
 
 				// see if open results hangs test
-				try (TupleQueryResult result = tupleQuery.evaluate()) {
-				}
+				// DO NOT CLOSE THIS
+				tupleQuery.evaluate();
```
Comment (jeenbroekstra) [inline testsuites/repository/src/main/java/org/eclipse/rdf4j/repository/TupleQueryResultTest.java:237]: Good find.

File: spring-components/pom.xml [hunk 2]
```diff
@@ -20,6 +20,13 @@
 	<description>Components to use with Spring</description>
 	<dependencyManagement>
 		<dependencies>
+			<dependency>
+				<groupId>org.junit</groupId>
+				<artifactId>junit-bom</artifactId>
+				<version>${junit.version}</version>
+				<type>pom</type>
```
Comment (jeenbroekstra) [inline spring-components/pom.xml:27]: Does this need to be included here if it's already in the root pom?

Comment (hmottestad) [reply spring-components/pom.xml:27]: The spring one overrides it.

**Pull request merged**